github/enterprise_billing_cost_centers.go

```diff
@@ -0,0 +1,232 @@
+// Copyright 2025 The go-github AUTHORS. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package github
+
+import (
+	"context"
+	"fmt"
+)
+
+// CostCenter represents an enterprise cost center.
+type CostCenter struct {
+	ID                string                `json:"id"`
+	Name              string                `json:"name"`
+	Resources         []*CostCenterResource `json:"resources"`
+	State             *string               `json:"state,omitempty"`
+	AzureSubscription *string               `json:"azure_subscription,omitempty"`
+}
+
+// CostCenterResource represents a resource assigned to a cost center.
+type CostCenterResource struct {
+	Type string `json:"type"`
+	Name string `json:"name"`
+}
+
+// CostCenters represents a list of cost centers.
+type CostCenters struct {
+	CostCenters []*CostCenter `json:"costCenters,omitempty"`
+}
+
+// CostCenterListOptions specifies optional parameters to the EnterpriseService.ListCostCenters method.
+type CostCenterListOptions struct {
```

Comment:
Maybe better:
```suggestion
type ListCostCenterOptions struct {
```

It's for consistency with:
https://github.com/google/go-github/blob/943b80101c7d15f44ae9e0caef2368b65f82e668/github/actions_artifacts.go#L59
https://github.com/google/go-github/blob/943b80101c7d15f44ae9e0caef2368b65f82e668/github/actions_workflow_runs.go#L61
https://github.com/google/go-github/blob/943b80101c7d15f44ae9e0caef2368b65f82e668/github/actions_runner_groups.go#L74
https://github.com/google/go-github/blob/943b80101c7d15f44ae9e0caef2368b65f82e668/github/actions_workflow_jobs.go#L60
https://github.com/google/go-github/blob/943b80101c7d15f44ae9e0caef2368b65f82e668/github/codespaces.go#L118